spec/index.md

```diff
@@ -1297,31 +1297,35 @@ Callbacks are addressed by their table index (as a proxy for a Wasm `funcref`).
 
 In the reply callback of a [inter-canister method call](#system-api-call), the argument refers to the response to that call. In reject callbacks, no argument is available.
 
+### Replicated and Non-Replicated execution mode
+
+Canister methods can be executed either in *replicated mode* where the method runs on all subnet nodes and the results go through consensus or in *non-replicated mode* where the method runs on a single node and the result does not go through consensus. Update methods can only be run in replicated mode (as they need to preserve state changes) while composite query methods can only be executed in non-replicated mode. Query methods can be executed in either mode.
```

Comment:
```suggestion
Canister methods can be executed either in *replicated mode* where the method runs on all subnet nodes and the results go through consensus or in *non-replicated mode* where the method runs on a single node and the result does not go through consensus. Update methods can only be run in replicated mode (as they need to preserve state changes) while composite query and inspect message methods can only be executed in non-replicated mode. Query methods can be executed in either mode.
```

Reply:
On the other hand, we aren't pointing out all replicated mode methods either, so I'm not sure about this suggestion after all.

Reply:
I think we should figure out what we want to call out here. We can list all methods and whether they can be run in replicated or non-replicated mode. I started with something to have a discussion.

Reply:
I added a table with all canister methods and the mode that they can be executed in. Was there anything else we said we wanted to include?

Reply:
We also discussed to include an overview of what changes get persisted in the individual cases, but we can also do that in a follow up MR to not block the improvement this MR already brings.

Reply:
Thanks for the reminder. I suggest we cover this in a follow up MR to unblock merging here.